src/NUnitFramework/tests/TestUtilities/TestDirectory.cs

```diff
@@ -42,6 +42,7 @@ public TestDirectory()
             Assume.That(_testDir.Exists, Is.False, _testDir + " should not already exist");
 
             _testDir.Create();
+            _testDir.Refresh();
             Assume.That(_testDir.Exists, Is.True, "Failed to create test dir " + _testDir);
```

Comment:
I think these are assumptions because we are not testing creating a test directory, that is just setup for another test. So, if say we don't have permissions to create the directory, then that other test hasn't failed, we just haven't met the prerequisites for running that test. Pretty fine line though, it could be either way :smile: